apps/emqx_bridge_rabbitmq/src/emqx_bridge_rabbitmq_source_worker.erl

```diff
@@ -0,0 +1,82 @@
+-module(emqx_bridge_rabbitmq_source_worker).
+
+-behaviour(gen_server).
+
+-export([start_link/1]).
+-export([
+    init/1,
+    handle_continue/2,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2
+]).
+
+-include_lib("amqp_client/include/amqp_client.hrl").
+
+start_link(Args) ->
+    gen_server:start_link(?MODULE, Args, []).
+
+init({_RabbitChannel, _InstanceId, _Params} = State) ->
+    {ok, State, {continue, confirm_ok}}.
+
+handle_continue(confirm_ok, State) ->
+    receive
+        #'basic.consume_ok'{} -> {noreply, State}
+    end.
+
+handle_call(_Request, _From, State) ->
+    {reply, ok, State}.
+
+handle_cast(_Request, State) ->
+    {noreply, State}.
+
+handle_info(
+    {#'basic.deliver'{delivery_tag = Tag}, #amqp_msg{
+        payload = Payload,
+        props = #'P_basic'{message_id = MessageId, headers = Headers}
+    }},
+    {Channel, InstanceId, Params} = State
+) ->
+    #{
+        hookpoints := Hooks,
+        payload_template := PayloadTmpl,
+        qos := QoSTmpl,
+        topic := TopicTmpl,
+        no_ack := NoAck
+    } = Params,
+    MQTTMsg = emqx_message:make(
+        make_message_id(MessageId),
+        InstanceId,
+        render(Payload, QoSTmpl),
```

Comment:
what happens if render fail?

Reply:
This function will not fail,  if it cannot find a placeholder, it will return <<"undefined">>.